malariagen_data/pf7.py

```diff
@@ -0,0 +1,278 @@
+import json
+import os
+
+import dask.array as da
+import pandas as pd
+import xarray
+import zarr
+
+from malariagen_data.util import (
+    DIM_ALLELE,
+    DIM_PLOIDY,
+    DIM_SAMPLE,
+    DIM_VARIANT,
+    da_from_zarr,
+    init_filesystem,
+    init_zarr_store,
+)
+
+DIM_ALT_ALLELE = "alt_alleles"
+DIM_STATISTICS = "sb_statistics"
+DIM_GENOTYPES = "genotypes"
+
+
+class Pf7:
+    def __init__(
+        self,
+        url,
+        data_config=None,
+        extended_calldata_variables=None,
+        extended_variant_fields=None,
```

Comment:
I'm not sure it's worth having these parameters. It would seem fine to me just to have a fixed set of extended fields, defined at the module level.

Reply:
I added these to make testing the extended function a bit easier. Instead of mocking the entire extended zarr I could just add a few extra variables. Similar to the `data_config` parameter, I thought most users wouldn't touch them, but more advanced users could specify the subset if they wanted to. Happy to change if you think this overcomplicates it thought

Reply:
I'd suggest to remove them, keep things simple. Would be good to have a conversation about the mocking tests, I'll put some thoughts about that in a separate comment.